deps/build.jl

```diff
@@ -200,7 +200,7 @@ PYTHONHOME = if !haskey(ENV, "PYTHONHOME") || use_conda
     exec_prefix = pysys(python, "exec_prefix")
     is_windows() ? exec_prefix : pysys(python, "prefix") * ":" * exec_prefix
 else
-    ENV["PYTHONHOME"]
+    get(ENV, "PYTHONHOME", nothing)
```

Comment:
Looking at what `Py_SetPythonHome` does after we first got this going, maybe it would be easier to do `get(ENV, "PYTHONHOME", "")` and possibly not need the rest of the patch after this point. Dunno whether a user having the env var `PYTHONHOME` set to an empty string is sane and should be honored.